task-sdk/tests/task_sdk/execution_time/test_task_runner.py

```diff
@@ -1928,6 +1928,83 @@ def execute(self, context):
         assert listener.error == error
 
 
+@pytest.mark.usefixtures("mock_supervisor_comms")
+class TestTaskRunnerCallsCallbacks:
+    def test_task_runner_calls_execute_callback(self):
+        results = []
+
+        def custom_callback(context):
+            results.append("callback")
+
+        class CustomOperator(BaseOperator):
+            def execute(self, context):
+                results.append("execute")
+
+        task = CustomOperator(
+            task_id="test_task_runner_calls_execute_callback",
+            on_execute_callback=custom_callback,
+        )
+        dag = get_inline_dag(dag_id="test_task_runner_calls_execute_callback", task=task)
+        ti = TaskInstance(
+            id=uuid7(),
+            task_id=task.task_id,
+            dag_id=dag.dag_id,
+            run_id="test_task_runner_calls_execute_callback",
+            try_number=1,
+        )
+
+        runtime_ti = RuntimeTaskInstance.model_construct(
+            **ti.model_dump(exclude_unset=True),
+            task=task,
+            start_date=timezone.utcnow(),
+        )
```

Comment:
You could use `create_runtime_ti` fixture

```suggestion
        runtime_ti = create_runtime_ti(dag_id="test_task_runner_calls_execute_callback", task=task)
```


https://github.com/apache/airflow/blob/f4af596beeed2a29fc4e878479f81b90ea8b140c/task-sdk/tests/conftest.py#L330